### PR TITLE
feature: lazy routing

### LINKED ISSUE #213: Split the app into modules
Split the app into modules like `Domains` module and make the `DomainsStoreProvider` available only under that module.

### PR DESCRIPTION
**Pull request by jurajpiar**
Creates a logical and functional routing structure that allows for loading providers only when required.

Fixes #213

### CODE REVIEW

**Comment by jurajpiar** [comment]
couldn't test storage done page because I'm getting no confirmations (new-block event received tho). Rns confirmations seem to work fine now.

**Comment by jurajpiar** [comment]
Not getting any storage confirmations though am getting new-block events. Rns ones are getting through just fine.
This means I couldn't test `STORAGE.DONE` path

**Comment by jurajpiar** [comment]
> LRGTM

Thanks did you manage to get confirmations for storage in the end?

**Comment by farinalvaro** [comment]
> Thanks did you manage to get confirmations for storage in the end?

@jurajpiar No, but I think it's a different PR. I accessed `storage/sell/done` directly and it's working 👍

File: src/components/pages/storage/StorageRoutes.tsx [hunk 1]
```diff
@@ -9,7 +9,21 @@ import { NotFound } from '..'
 import StorageLandingPage from './StorageLandingPage'
 import StorageOffersPage from './buy/StorageOffersPage'
 import StorageListingPage from './sell/StorageListingPage'
-import StorageOfferListed from './sell/StorageOfferListed'
```
Comment (farinalvaro) [inline src/components/pages/storage/StorageRoutes.tsx:12]: We should render this component on STORAGE.SELL.DONE

File: src/components/pages/storage/StorageRoutes.tsx [hunk 2]
```diff
@@ -9,7 +9,21 @@ import { NotFound } from '..'
...
+    <Route exact path={ROUTES.STORAGE.BUY.BASE} component={StorageOffersPage} />
+  </Switch>
+)
+
+const SellingRoutes = () => (
+  <Switch>
+    <StorageListingStoreProvider>
+      <Route exact path={ROUTES.STORAGE.SELL.BASE} component={StorageListingPage} />
+    </StorageListingStoreProvider>
+    <Route exact path={ROUTES.STORAGE.SELL.DONE} component={StorageOffersPage} />
```
Comment (farinalvaro) [inline src/components/pages/storage/StorageRoutes.tsx:24]: We need to put this one on top or wrap the previous one

File: src/components/pages/storage/StorageRoutes.tsx [hunk 3]
```diff
@@ -11,20 +11,32 @@ import StorageOffersPage from './buy/StorageOffersPage'
...
+const BuyingRoutes = () => (
+  <Switch>
+    <Route exact path={ROUTES.STORAGE.BUY.BASE} component={StorageOffersPage} />
+  </Switch>
+)
+
+const SellingRoutes = () => (
+  <Switch>
+    <Route exact path={ROUTES.STORAGE.SELL.DONE} component={StorageOfferListed} />
+    <StorageListingStoreProvider>
```
Comment (farinalvaro) [inline src/components/pages/storage/StorageRoutes.tsx:23]: I think we also need to add `<Route component={NotFound} />` for the routes that are `storage/sell/x`, before the provider

File: src/components/pages/storage/StorageRoutes.tsx [hunk 4]
```diff
@@ -18,14 +18,23 @@ const StorageRoutes = () => {
   if (storageEnabled) {
     return (
       <Switch>
-        <Redirect exact from={ROUTES.STORAGE.BASE} to={ROUTES.STORAGE.BUY} />
+        <Redirect exact from={ROUTES.STORAGE.BASE} to={ROUTES.STORAGE.BUY.BASE} />
         <Route exact path={ROUTES.STORAGE.BASE} component={StorageLandingPage} />
```
Comment (farinalvaro) [inline src/components/pages/storage/StorageRoutes.tsx:22]: If storage is enabled, we shouldn't show the landing page

File: src/components/pages/storage/StorageRoutes.tsx [hunk 5]
```diff
@@ -18,14 +18,23 @@ const StorageRoutes = () => {
...
+          </Switch>
+        </Route>
+        <Route path={ROUTES.STORAGE.SELL.BASE}>
+          <StorageListingStoreProvider>
+            <Route exact path={ROUTES.STORAGE.SELL.BASE} component={StorageListingPage} />
+          </StorageListingStoreProvider>
+          <Switch>
+            <Route exact path={ROUTES.STORAGE.SELL.DONE} component={StorageOfferListed} />
+            <Route component={NotFound} />
+          </Switch>
```
Comment (farinalvaro) [inline src/components/pages/storage/StorageRoutes.tsx:36]: I'm getting the 404 page under `storage/sell`, I think we should wrap the 3 routes inside a common switch, may be we need to wrap the listing page with the store
Screen Shot 2020-08-12 at 09 52 53

Comment (jurajpiar) [reply src/components/pages/storage/StorageRoutes.tsx:36]: Ah I missed this too. Can't wrap it under the same switch, at least not directly as switch can only contain route or redirect

**Pull request merged**